presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -152,7 +156,31 @@ private Expression simplifyExpression(Expression expression)
             expression = ExpressionTreeRewriter.rewriteWith(new ExtractCommonPredicatesExpressionRewriter(), expression, NodeContext.ROOT_NODE);
             IdentityLinkedHashMap<Expression, Type> expressionTypes = getExpressionTypes(session, metadata, sqlParser, types, expression, emptyList() /* parameters already replaced */);
             ExpressionInterpreter interpreter = ExpressionInterpreter.expressionOptimizer(expression, metadata, session, expressionTypes);
-            return LiteralInterpreter.toExpression(interpreter.optimize(NoOpSymbolResolver.INSTANCE), expressionTypes.get(expression));
+            expression = LiteralInterpreter.toExpression(interpreter.optimize(NoOpSymbolResolver.INSTANCE), expressionTypes.get(expression));
+            expression = ExpressionTreeRewriter.rewriteWith(new TransformSingleValueInPredicateRewriter(), expression);
+            return expression;
+        }
+    }
+
+    private static class TransformSingleValueInPredicateRewriter
```

Comment:
I implemented this optimization as an optimizer/rule that rewrites the plan tree because it may reveal some opportunity for further optimizations. But I agreed that we should keep scalar expression optimizations in one place.